requires.go

```diff
@@ -0,0 +1,33 @@
+package task
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/go-task/task/v3/taskfile"
+)
+
+func (e *Executor) areTaskRequiredVarsSet(ctx context.Context, t *taskfile.Task, call taskfile.Call) (bool, error) {
+	if len(t.Requires) == 0 {
+		return true, nil
+	}
+
+	vars, err := e.Compiler.GetVariables(t, call)
+	if err != nil {
+		return false, err
+	}
+
+	missingVars := []string{}
+	for _, requiredVar := range t.Requires {
+		if !vars.Exists(requiredVar) {
+			missingVars = append(missingVars, requiredVar)
+		}
+	}
+
+	if len(missingVars) > 0 {
+		return false, errors.New("required variables not set: " + strings.Join(missingVars, ","))
```

Comment:
We prefix all error messages with `task:` to differentiate them from errors from other CLI tools:

```suggestion
		return false, errors.New("task: required variables not set: " + strings.Join(missingVars, ","))
```

Reply:
It might also be nice to add a custom error code for this. [See here](https://github.com/go-task/task/blob/main/errors/errors_task.go) for examples.

Reply:
Custom error & code added

Reply:
I should've mentioned this before. Could we also add the new error code to the [API reference doc](https://github.com/go-task/task/blob/main/docs/docs/api_reference.md#exit-codes).

Reply:
Code added to API doc